FatPkg/EnhancedFatDxe/Init.c

```diff
@@ -61,6 +61,30 @@ FatAllocateVolume (
   //
   Volume->RootDirEnt.FileString       = Volume->RootFileString;
   Volume->RootDirEnt.Entry.Attributes = FAT_ATTRIBUTE_DIRECTORY;
+
+  //
+  // Check to see if the underlying block device's BlockSize meets what the FAT spec requires
+  //
+  if ((BlockIo == NULL) || (BlockIo->Media == NULL)) {
+    DEBUG ((DEBUG_ERROR, "%a BlockIo or BlockIo is NULL!\n", __func__));
+    Status = EFI_INVALID_PARAMETER;
+    goto Done;
+  }
+
+  if ((BlockIo->Media->BlockSize > (1 << MAX_BLOCK_ALIGNMENT)) ||
+      (BlockIo->Media->BlockSize < (1 << MIN_BLOCK_ALIGNMENT)))

```

Comment:
Does BlockSize have to be a power of 2 in that range of supported block sizes?

Reply:
I will make this explicit. I think any current device has a power of 2 block size, but there are only four supported block sizes for FAT, so it is not onerous to check explicitly.

Reply:
I agree with Mike. Can use IS_POW2 to check.